internal/harness/golang/matchfunction_service.go

```diff
@@ -74,6 +73,7 @@ type MatchFunctionParams struct {
 	Properties        *structpb.Struct
 	Rosters           []*pb.Roster
 	PoolNameToTickets map[string][]*pb.Ticket
+	FunctionName      string
```

Comment:
Why is this being done? The MMF invokes the harness - supplying the name in FunctionSettings - and we channel that same name back through he params for the MMF to use in generating matches? Doesnt the mmf already know its name?

Reply:
Also, please separate out unrelated changes into separate PRs - This PR description is for running e2e with random ports  - so lets just stick to the changes relevant for that in this PR. Please remove any improvements to other bits in the code (like mmf examples etc.) to a separate PR.